app/models/components/course/duplicator_component_host.rb

```diff
@@ -0,0 +1,15 @@
+class Duplicator::ModelComponentHost
```

Comment:
I was thinking that since a course can also be duplicated, the framework for duplicating it should be on a higher level, outside of the course itself.

Of course, if that isn't the case, I'm ok with putting it in with the course model host, either in the same file or as a mixin.